contrib/epee/src/readline_buffer.cpp

```diff
@@ -16,6 +17,8 @@ static std::string last_prompt;
 std::mutex line_mutex, sync_mutex, process_mutex;
 std::condition_variable have_line;
 
+std::vector<std::string> rdln::readline_buffer::completion_commands = {"exit"};
```

Comment:
I was not suggesting that you remove statics, just that to re-organize them into one class that was declared statically within a function. The class would contain the entire "state" of readline, but not necessarily any of the helper functions you have provided.

The major part part of the suggestion was a move from namespace statics to function statics. Function statics have no issues if accessed before main during the namespace initialization stage. While this is currently not an issue, this type of static usage is known to cause problems as codebase is being maintained. FWIW I am not particularly worried about it since only readline is ever likely to call these functions, but I will cleanup this sort of static usage if it is related to something I working on.